### ISSUE TITLE
Error: " Build Warning: Layout 'default' requested in ... does not exist."

### ISSUE THREAD

**Issue by tomjohnson1492** [slashdotdash/jekyll-lunr-js-search#63]
When I build my site, I'm getting all kinds of warning errors about default.html layout not found:

```
 Generating...
              Lunr: Creating search index...
     Build Warning: Layout 'default' requested in content/adtruth/archive.html does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/archive_getting_started.html does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/demo.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/demo_environment.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/faqs.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/hello_world.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/identifiers.md does not exist.
     Build Warning: Layout 'default' requested in index.html does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/make_requests.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/matchdevice.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/overview.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/questions.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/release_notes.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/rulesets.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/sample_scenario.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/scrollspy.md does not exist.
     Build Warning: Layout 'default' requested in search.html does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/troubleshooting_hmac.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/use_cases.md does not exist.
     Build Warning: Layout 'default' requested in content/adtruth/windows.md does not exist.
     Build Warning: Layout 'post' requested in _posts/2014-06-28-example-bootstrap.md does not exist.
     Build Warning: Layout 'post' requested in _posts/2014-07-27-exmaple-css.md does not exist.
     Build Warning: Layout 'post' requested in _posts/2014-07-28-introducing-shiori.md does not exist.
              Lunr: Index ready (lunr.js v0.4.5)
```

Removing the search plugin made these errors go away. Do you know why the errors are happening? They didn't occur in my plain, simple search integration site.

**Comment by Ray-Cheng** [comment]
@tomjohnson1492 ,check this : #34 ,https://github.com/slashdotdash/jekyll-lunr-js-search/commit/11156ef5aa3170a13a1415d045fae1fe56c44e4d

**Comment by tomjohnson1492** [comment]
Thanks, but those threads didn't help much. #34 is inconclusive. 11156ef is vague. Am I supposed to make those changes in the plugin source files and rebuild it or something?

**Comment by Ray-Cheng** [comment]
@tomjohnson1492,yes change source as #11156ef,then jekyll build,jekyll server.

**Comment by llbbl** [comment]
I commented in #34 ... I was able to fix it using the suggested change, but not sure if it works with the current version of the plugin. Commenting here also because this thread lead me to the fix.

**Comment by slashdotdash** [comment]
I've just published an update to the plugin and gem that should fix this error. When extracting the content of documents for the search index the layout is unset.

    gem install jekyll-lunr-js-search -v3.0.0

**Comment by amine8ghandi8amine** [comment]
ERROR:  Error installing jekyll-lunr-js-search:
	ERROR: Failed to build gem native extension.

    /usr/bin/ruby2.1 extconf.rb
mkmf.rb can't find header files for ruby at /usr/lib/ruby/include/ruby.h

extconf failed, exit code 1

Gem files will remain installed in /var/lib/gems/2.1.0/gems/nokogiri-1.6.8 for inspection.
Results logged to /var/lib/gems/2.1.0/extensions/x86_64-linux/2.1.0/nokogiri-1.6.8/gem_make.out

**Comment by shane-axiom** [comment]
Still seeing this in jekyll-lunr-js-search 3.2.0.

**Comment by bjgordon** [comment]
I was having the error with 3.2.0, jekyll 3.3.0. This made the warning go away for me: 

Change [this line](https://github.com/slashdotdash/jekyll-lunr-js-search/blob/master/lib/jekyll_lunr_js_search/page_renderer.rb#L14)
from `item.data.delete("layout")`
to `item.data["layout"] = nil`

**Comment by nkuehn** [comment]
@bjgordon @slashdotdash  the `item.data["layout"] = nil` variant is the correct and working one for all users since you need  leave an explicit value for the layout to prevent jekyll from falling back to the default layout configured in the front matter defaults in _config.yml 

The current approach with removing the property completely was probably not tested in a project that uses a default layout in front matter defaults.

**Comment by slashdotdash** [comment]
I've just merged pull request #109 with that change.

**Comment by menehune23** [comment]
Is this in the latest release? I'm seeing this in 3.3.0.
